src/plugins/instrument_fetch.js

```diff
@@ -227,50 +231,27 @@ class InstrumentFetch {
 
     _shouldTrace(tracer, url) {
         // This shouldn't be possible, but let's be paranoid
-        if (!tracer) {
+        if (!tracer || !url) {
             return false;
         }
 
         let opts = tracer.options();
         if (opts.disabled) {
             return false;
         }
-        if (!url) {
+
+        if (this._internalExclusions.some(ex => ex.test(url))) {
```

Comment:
I wasn't sure why these were `for ... in` loops. They're arrays, right? So we shouldn't use the slower and more error-prone for in loop here. But perhaps I missed a constraint?

Reply:
This optimization looks much cleaner to me 00. I don't think you're missing anything:? Minor addition is I do think that for the `inclusion-pattern` check your argument should probably be `inc` instead of `ex`

Reply:
Done!